attestation-service/src/verifier/cca/mod.rs

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2023 Arm Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::*;
+use anyhow::{anyhow, Context, Result};
+use async_trait::async_trait;
+use core::result::Result::Ok;
+use serde::{Deserialize, Serialize};
+use serde_json::{Map, Value};
+use veraison_apiclient::*;
+
+const VERIFIER_ADDR: &str = "VERIFIER_ADDR";
+const DEFAULT_VERIFIER_ADDR: &str = "localhost:8080";
+const MEDIA_TYPE: &str = "application/eat-collection; profile=http://arm.com/CCA-SSD/1.0.0";
+
+#[derive(Debug, Default)]
+pub struct CCA {}
+
+#[derive(Serialize, Deserialize)]
+struct CcaEvidence {
+    /// CCA token
+    token: Vec<u8>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct RealmToken {
+    //cca_realm_personalization_value: String,
+    cca_realm_initial_measurement: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "kebab-case")]
+struct Evidence {
+    cca_realm_delegated_token: RealmToken,
+}
+
+fn my_evidence_builder(
+    nonce: &[u8],
+    accept: &[String],
+    token: Vec<u8>,
+) -> Result<(Vec<u8>, String), veraison_apiclient::Error> {
+    log::info!("server challenge: {:?}", nonce);
+    log::info!("acceptable media types: {:#?}", accept);
+    // TODO: Get the CCA media type from the slice of `accept`.
+    Ok((token, MEDIA_TYPE.to_string()))
+}
+
+#[async_trait]
+impl Verifier for CCA {
+    async fn evaluate(
+        &self,
+        _nonce: String,
```

Comment:
this is not renamed as `_nonce`,  pls expect a follow up pr to address the `nonce` comparison.

Reply:
Ok, could you please make an issue about this and add notes inside the code to refer to the issue?

Reply:
Done